### PR TITLE
Adding 1 more Security Baseline check (55 total so far), expanding remediation for 15 checks, and other fixes following validation via Azure Policy

### PR DESCRIPTION
**Pull request by MariusNi**
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] I added unit-tests to validate my changes. All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/common/commonutils/DeviceInfoUtils.c [hunk 1]
```diff
@@ -287,8 +287,13 @@ bool IsCpuFlagSupported(const char* cpuFlag, void* log)
 
     if ((NULL != cpuFlag) && (NULL != strstr(cpuFlags, cpuFlag)))
     {
+        OsConfigLogInfo(log, "CPU flag'%s' is supported", cpuFlag);
```
Comment (robertschaedler3) [inline src/common/commonutils/DeviceInfoUtils.c:290]: nit: should there be a space here: `flag '%s'`

**Review by robertschaedler3 — APPROVED** [review]
👍

File: src/common/commonutils/DeviceInfoUtils.c (hunk 1 above)
Comment (MariusNi) [reply src/common/commonutils/DeviceInfoUtils.c:290]: Good catch

**Review by robertschaedler3 — DISMISSED** [review]
👍

**Review by simonjaeger — APPROVED** [review]
:shipit:

**Review by robertschaedler3 — APPROVED** [review]
👍

**Review by simonjaeger — APPROVED** [review]
:shipit:

**Pull request merged**